packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -278,9 +278,23 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
   const updateRenderContext = React.useCallback(
     (nextRenderContext) => {
       setRenderContext(nextRenderContext);
+
+      const [firstRowToRender, lastRowToRender] = getRenderableIndexes({
+        firstIndex: nextRenderContext.firstRowIndex,
+        lastIndex: nextRenderContext.lastRowIndex,
+        minFirstIndex: 0,
+        maxLastIndex: currentPage.rows.length,
+        buffer: rootProps.rowBuffer,
+      });
+
+      apiRef.current.publishEvent('renderedRowsIntervalChange', {
```

Comment:
We already calculate `firstRowToRender` inside `getRows`. Would it work if we change to publish the event inside this method? You could use a ref to keep track of the last render context, to avoid publishing the event when nothing has changed.

The `buffer` param in `getRows` also considers if virtualization is disabled. This may not be relevant here but reduces duplicated code.

Reply:
Doesn't this introduce additional complexity? I remember that I initially publish the event from `getRows` but then it fired more ofthen than needed so I adjusted it to be fired only when the context changes and this seemed the correct place to do it from, no?